services/packages/packages.go

```diff
@@ -657,3 +659,35 @@ func RemoveAllPackages(ctx context.Context, userID int64) (int, error) {
 	}
 	return count, nil
 }
+
+func LinkPackageToRepository(ctx context.Context, doer *user_model.User, p *packages_model.Package, repoID int64) error {
+	if repoID != 0 {
+		repo, err := repo_model.GetRepositoryByID(ctx, repoID)
+		if err != nil {
+			return fmt.Errorf("Error getting repository %d: %w", repoID, err)
+		}
+
+		canWrite := repo.OwnerID == doer.ID
+
+		if !canWrite {
+			perms, err := access_model.GetUserRepoPermission(ctx, repo, doer)
+			if err != nil {
+				return fmt.Errorf("Error getting repository permissions for %d on %d: %w", doer.ID, repo.ID, err)
+			}
+
+			canWrite = perms.CanWrite(unit.TypePackages)
+		}
+
+		if !canWrite {
+			return fmt.Errorf("No permissions to link this package and repository")
+		}
```

Comment:
```suggestion
	  perms, err := access_model.GetUserRepoPermission(ctx, repo, doer)
	  if err != nil {
		  return fmt.Errorf("Error getting repository permissions for %d on %d: %w", doer.ID, repo.ID, err)
	  }

		if !perms.CanWrite(unit.TypePackages) {
			return fmt.Errorf("No permissions to link this package and repository")
		}
```

It is also possible that packages are disabled for this repo, so I think we should check for all users, not just if user isn't the owner.

		
```